subprojects/emcee/queue-worker-api/src/main/kotlin/com/avito/emcee/worker/GetBucketResponse.kt

```diff
@@ -0,0 +1,31 @@
+package com.avito.emcee.worker
+
+import com.avito.emcee.queue.BuildArtifacts
+import com.avito.emcee.queue.Device
+import com.avito.emcee.queue.TestEntry
+import com.avito.emcee.queue.TestExecutionBehavior
+import com.avito.emcee.queue.TestTimeoutConfiguration
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+import dev.zacsweers.moshix.sealed.annotations.TypeLabel
+
+@JsonClass(generateAdapter = true, generator = "sealed:caseId")
```

Comment:
@dsvoronin copy-pasting your code

Reply:
https://github.com/ZacSweers/MoshiX/tree/main/moshi-sealed
i'm using this approach